dask/dataframe/shuffle.py

```diff
@@ -744,18 +760,28 @@ def collect(p, part, meta, barrier_token):
         return res if len(res) > 0 else meta
 
 
-def set_partitions_pre(s, divisions):
+def set_partitions_pre(s, divisions, ascending=True):
     try:
-        partitions = divisions.searchsorted(s, side="right") - 1
+        if ascending:
+            partitions = divisions.searchsorted(s, side="right") - 1
+        else:
+            partitions = -divisions.searchsorted(s, side="right") % (len(divisions) - 1)
```

Comment:
It took me a bit to figure out what this is doing. This seems easier to understand and I think it's equivalent in this case:

```suggestion
            partitions = len(divisions) - 1 - divisions.searchsorted(s, side="right")
```

Reply:
Thanks @jsignell! Yeah that expression is a lot simpler and seems to accomplish the same thing